Gruntfile.js

```diff
@@ -0,0 +1,112 @@
+module.exports = function (grunt) {
+	grunt.initConfig({
+		ts: {
+			options: grunt.file.readJSON("tsconfig.json").compilerOptions,
+
+			devlib: {
+				src: ["lib/**/*.ts"],
+				reference: "lib/.d.ts"
+			},
+
+			devall: {
+				src: ["lib/**/*.ts", "test/**/*.ts", "definitions/**/*.ts"],
+				reference: "lib/.d.ts"
+			},
+
+			release_build: {
+				src: ["lib/**/*.ts", "test/**/*.ts"],
+				reference: "lib/.d.ts",
+				options: {
+					sourceMap: false,
+					removeComments: true
+				}
+			}
+		},
+
+		tslint: {
+			build: {
+				files: {
+					src: ["lib/**/*.ts", "test/**/*.ts", "definitions/**/*.ts", "!**/*.d.ts"]
+				},
+				options: {
+					configuration: grunt.file.readJSON("./tslint.json")
+				}
+			}
+		},
+
+		watch: {
+			devall: {
+				files: ["lib/**/*.ts", 'test/**/*.ts'],
+				tasks: [
+					'ts:devall',
+					'shell:npm_test'
+				],
+				options: {
+					atBegin: true,
+					interrupt: true
+				}
+			},
+			ts: {
+				files: ["lib/**/*.ts", "test/**/*.ts"],
+				tasks: [
+					'ts:devall'
+				],
+				options: {
+					atBegin: true,
+					interrupt: true
+				}
+			}
+		},
+
+		shell: {
+			options: {
+				stdout: true,
+				stderr: true,
+				failOnError: true
+			},
+			build_package: {
+				command: "npm pack"
+			},
+			npm_test: {
+				command: "npm test"
+			}
+		},
+
+		clean: {
+			src: ["test/**/*.js*", "lib/**/*.js*", "!lib/hooks/**/*.js", "!Gruntfile.js", "*.tgz"]
+		}
+	});
+
+	grunt.loadNpmTasks("grunt-contrib-clean");
+	grunt.loadNpmTasks("grunt-contrib-watch");
+	grunt.loadNpmTasks("grunt-shell");
+	grunt.loadNpmTasks("grunt-ts");
+	grunt.loadNpmTasks("grunt-tslint");
+
+	grunt.registerTask("delete_coverage_dir", function () {
+		var done = this.async();
+		var rimraf = require("rimraf");
+		rimraf("coverage", function (err) {
```

Comment:
what is rimraf? We do not have it as dependency or devDependency (more applicable)